test/nemsfv3gfs/regression_test_nemsfv3gfs.py

```diff
@@ -0,0 +1,228 @@
+###############################################################################
+
+def parse_arguments():
+    """Parse command line arguments."""
+    description = 'Regression test / code checkout script for NEMSfv3gfs'
+    parser = argparse.ArgumentParser(description=description)
+    parser.add_argument('--config', action='store', help='name of regression test config', 
+                        default='regression_test_nemsfv3gfs.cfg')
+    args = parser.parse_args()
+    return args.config
+
+def parse_config(configfile):
+    """Parse the specified config file and return configuration."""
+    if os.path.exists(configfile):
+        config = ConfigParser.RawConfigParser()
+        config.read(configfile)
+    else:
+        raise Exception('Config file {0} not found'.format(configfile))
+    # Base directory
+    basedir = config.get('default', 'basedir')
+    # Dictionaries with names of forks and branches for all repositories
+    forks = {}
+    branches = {}
+    for repository in REPOSITORIES:
+        forks[repository] = config.get(repository, 'fork')
+        branches[repository] = config.get(repository, 'branch')
+    # Flag whether to run the tests or not (i.e. check out code only)
+    runtests = config.getboolean('testing', 'runtests')
+    # Account name/number for the scheduler
+    account = config.get('testing', 'account')
+    # Compiler to use when running the tests
+    compiler = config.get('testing', 'compiler')
+    # Test configuration
+    testconfig = config.get('testing', 'config')
+    return (basedir, forks, branches, runtests, account, compiler, testconfig)
+
+def execute(cmd, abort = True):
+    """Runs a local command in a shell and capture the exit status.
+    If abort = True, abort in case an error occurs during the
+    execution of the command."""
+    print "Executing '{0}'".format(cmd)
+    status = os.system(cmd)
+    if not status == 0:
+        if abort:
+            message = 'Execution of command {0} failed, return code = {1}'.format(cmd, status)
+            raise Exception(message)
+        else:
+            message = 'WARNING: Execution of command {0} failed, return code = {1}; ignore and proceed'.format(cmd, status)
+            print message
+    return status
+
+def setup_workdir(basedir):
+    """Sets up the work directory for local clone of repositories
+    underneath the base directory. Aborts if the base directory
+    does not exist, deletes existing work directories. The work
+    subdirectory is labeled as rt_YYYYMMDDTHHMMSS."""
+    if not os.path.exists(basedir):
+        raise Exception('Base directory {0} does not exist'.format(basedir))
+    now = datetime.datetime.now()
+    workdir = os.path.abspath(os.path.join(basedir, 'rt_{0}'.format(now.strftime('%Y%m%dT%H%M%S'))))
+    print "Setting up work directory {0} for local clone of repositories".format(basedir)
+    if os.path.exists(workdir):
+        shutil.rmtree(workdir)
+    os.makedirs(workdir)
```

Comment:
The script in its current version is not meant to be reused if a test fails. In general, running the regression tests for NEMSfv3gfs is a monolithic system, and any shortcutting requires way more knowledge than what we can expect from others.

That said, I will provide instructions on how to execute the rt.sh script manually and run only certain tests, which will require the user to go the the respective submodule (in your case ccpp-framework) and update the fork from remote (since the remote is configured, that's straightforward).